packages/orchestration/src/exos/cosmos-orchestration-account.js

```diff
@@ -1128,6 +1140,48 @@ export const prepareCosmosOrchestrationAccountKit = (
             watch(E(this.facets.helper.owned()).executeEncodedTx(msgs, opts)),
           );
         },
+        /** @type {HostOf<NobleMethods['depositForBurn']>} */
+        depositForBurn(destination, amount) {
+          return asVow(() => {
+            trace('depositForBurn', { destination, amount });
+            const { helper } = this.facets;
+            const { chainAddress } = this.state;
+
+            /** @type {EthChainInfo} */
+            // @ts-expect-error cast
+            const chainInfo = chainHub.getChainInfoByChainId(
+              destination.chainId,
+            );
+            typeof chainInfo.cctpDestinationDomain === 'number' ||
+              Fail`Must be reachable by CCTP`;
+            const { cctpDestinationDomain } = chainInfo;
+
+            if (typeof cctpDestinationDomain !== 'number') {
+              throw Fail`${q(destination.chainId)} does not have "cctpDestinationDomain" set in ChainInfo`;
+            }
+
+            // see https://github.com/circlefin/noble-cctp/blob/master/examples/depositForBurn.ts#L52-L70
+            const mintRecipient = leftPadEthAddressTo32Bytes(destination.value);
```

Comment:
This also needs to support solana, please add a TODO.

When this is implemented, we'll probably want to look for `chainInfo.ecosystem` to determine which address encoding function we need to use.